### ISSUE TITLE
Section 2.2: Why fit entire dataset to tokenizer?

### ISSUE THREAD

**Issue by teohsinyee** [chriskhanhtran/CNN-Sentence-Classification-PyTorch#1]
Hi @chriskhanhtran , thanks for efforts to make this organized repo. I've an inquiry about Section 2.2 for the codes.
Referring to the code below, `texts` is the entire dataset. Which means we here fit our tokenizer with entire dataset. 
```
tokenized_texts, word2idx, max_len = tokenize(texts)
input_ids = encode(tokenized_texts, word2idx, max_len)
```
I doubt will this biasing our model since test set is included.
However, isn't better if we just fit the Training set only to the Tokenizer? 
Which we split the data first before doing tokenization.

Reference: https://stackoverflow.com/a/54892165/12307615
